### PR TITLE
Store x and y drifts in S3 and S4

### PR DESCRIPTION
**Pull request by cpiaulet**
- Stage 3: loop over integrations to obtain the exact position of the trace along the spatial dimension by e.g. fitting a Gaussian (record mean and std) to the histogram of fluxes along the spatial axis -> saved to the Spec file + diagnostics plots
- Stage 4: save the y position and width information from Stage 3 to the LCData and record the drifts in the x (wavelength/dispersion) direction and the width of a Gaussian fitted to the CCF of each integration, cross-correlated with a reference spectrum (e.g. the last) within the same data file. These x positions (drifts) and their widths are saved to the Spec and the LCData files.

### CODE REVIEW

**Comment by taylorbell57** [comment]
Hi Caroline, I see you accidentally uploaded a lot of analysis files by accident - can you please delete them from this PR? In general, I recommend you store your Eureka! outputs somewhere outside of the Eureka folder to avoid such issues in the future

**Comment by cpiaulet** [comment]
are things alright now @taylorbell57 ?

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/plots_s3.py:347]: This should be figure 3305

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/plots_s3.py:381]: This should be figure 3306

File: src/eureka/S3_data_reduction/s3_reduce.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/s3_reduce.py:366]: Instead of overwriting `meta.src_ypos` here which might have unintended consequences, use just a `_` instead to discard it

File: src/eureka/S3_data_reduction/s3_reduce.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/s3_reduce.py:268]: Your `use_hdr` isn't going to be needed once PR #373 gets merged

File: src/eureka/S3_data_reduction/s3_reduce.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/s3_reduce.py:365]: Same comment here about `use_hdr`

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/plots_s3.py:370]: You shouldn't need to do `np.array` - if anything you can do `data["driftypos"].values` if just `data["driftypos"]` doesn't work

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/plots_s3.py:404]: Same comment about `np.array` as above

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/source_pos.py:19]: `integ : int, optional`

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/source_pos.py:59]: Once PR #373 gets merged, this will need to be changed to just
```
if meta.src_pos_type == 'gaussian':
```

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/source_pos.py:76]: `integ : int, optional`

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/source_pos.py:127]: `integ : int, optional`

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/source_pos.py:213]: `integ : int, optional`

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/source_pos.py:48]: A PSF width can also be estimated using a FWM-based method. As an example, see SPCA's centroid_FWM function in Photometry_Aperture.py [link](https://github.com/lisadang27/SPCA/blob/30885e1443ce8957c6280efb5846be8caad6a9b2/SPCA/Photometry_Aperture.py#L63) where xw and yw are estimates for the PSF width for Spitzer's photometric data

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (taylorbell57) [inline src/eureka/S4_generate_lightcurves/s4_genLC.py:125]: This should remain an empty line

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (taylorbell57) [inline src/eureka/S4_generate_lightcurves/s4_genLC.py:200]: This should remain an empty line

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (taylorbell57) [inline src/eureka/S4_generate_lightcurves/s4_genLC.py:237]: I would instead write 'Computing drift/jitter...'

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (taylorbell57) [inline src/eureka/S4_generate_lightcurves/s4_genLC.py:229]: For now, to avoid breaking backwards compatibility I would write:
```
if hasattr(meta, 'record_ypos') and meta.record_ypos:
```

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (taylorbell57) [inline src/eureka/S4_generate_lightcurves/s4_genLC.py:253]: I personally wouldn't bother putting these in spec - thoughts @kevin218?

File: demos/JWST/S3_miri_lrs_template.ecf
Comment (taylorbell57) [inline demos/JWST/S3_miri_lrs_template.ecf:23]: Please document this new ECF setting at docs/source/ecf.rst

File: demos/JWST/S4_template.ecf
Comment (taylorbell57) [inline demos/JWST/S4_template.ecf:12]: Please document this new ECF setting at docs/source/ecf.rst

**Review by taylorbell57 — CHANGES_REQUESTED** [review]
Just a few minor tweaks to suggest. I also think it is worth waiting until my PR #373 is merged before merging this PR (because of the conflicts surrounding the source_pos function)

**Comment by taylorbell57** [comment]
Please also add the new ECF settings to the WFC3 test ECFs to fix those tests

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/plots_s3.py]: done

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/plots_s3.py]: done

File: src/eureka/S3_data_reduction/s3_reduce.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/s3_reduce.py]: done

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/plots_s3.py]: done

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/plots_s3.py]: done

File: src/eureka/S3_data_reduction/source_pos.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/source_pos.py]: done

File: src/eureka/S3_data_reduction/source_pos.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/source_pos.py]: done

File: src/eureka/S3_data_reduction/source_pos.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/source_pos.py]: done

File: src/eureka/S3_data_reduction/source_pos.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/source_pos.py]: done

File: src/eureka/S3_data_reduction/source_pos.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/source_pos.py:48]: I am not sure what this means here for my PR, although I know the method - can you clarify what this means in terms of modifications?

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (cpiaulet) [reply src/eureka/S4_generate_lightcurves/s4_genLC.py]: done

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (cpiaulet) [reply src/eureka/S4_generate_lightcurves/s4_genLC.py:200]: done

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (cpiaulet) [reply src/eureka/S4_generate_lightcurves/s4_genLC.py]: done

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (cpiaulet) [reply src/eureka/S4_generate_lightcurves/s4_genLC.py]: done

File: src/eureka/S4_generate_lightcurves/s4_genLC.py
Comment (cpiaulet) [reply src/eureka/S4_generate_lightcurves/s4_genLC.py:253]: I would like to have them in spec, because people that do not want to use Eureka's S4 binned LC output and do their own binning will still want to have the x drifts recorded - for detrending purposes.

File: demos/JWST/S3_miri_lrs_template.ecf
Comment (cpiaulet) [reply demos/JWST/S3_miri_lrs_template.ecf:23]: done

File: demos/JWST/S4_template.ecf
Comment (cpiaulet) [reply demos/JWST/S4_template.ecf:12]: done

File: src/eureka/S3_data_reduction/s3_reduce.py
Comment (cpiaulet) [reply src/eureka/S3_data_reduction/s3_reduce.py:364]: done

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/plots_s3.py:389]: You'll need to update this filename

File: src/eureka/S3_data_reduction/plots_s3.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/plots_s3.py:422]: And this file name as well

File: src/eureka/S3_data_reduction/source_pos.py
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/source_pos.py:50]: Let's do this in a future PR, but I just meant that it would be nice to be able to get a src_ywidth returned out of this function if one was to use the FWM method instead of Gaussian.

**Comment by taylorbell57** [comment]
Also, please address any non-NIRISS flake8 issues that remain - I see there are a couple left

File: src/eureka/S3_data_reduction/s3_reduce.py
Comment (taylorbell57) [inline src/eureka/S3_data_reduction/s3_reduce.py:364]: header is no longer a function input

**Comment by taylorbell57** [comment]
Oh, also please update the S3 and S4 ECFs in docs/media

**Pull request merged**
